binaries/coordinator/src/lib.rs

```diff
@@ -675,9 +675,10 @@ async fn handle_destroy(
         )
         .await?;
     }
-    destroy_daemons(daemon_connections, clock.new_timestamp()).await?;
+
+    let result = destroy_daemons(daemon_connections, clock.new_timestamp()).await;
     *daemon_events_tx = None;
-    Ok(())
+    result
```

Comment:
This is the most relevant part, right? I.e. that we set `daemon_events_tx` to `None` even if there were errors?

Reply:
Yes, this will end the main loop. 

I had this issue more often on slower machine or remote machine.